src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -245,21 +245,25 @@ else if (temporalIdx > 0)
     private static List<SecondarySsrc> getSecondarySsrcs(long ssrc, List<SourceGroupPacketExtension> sourceGroups)
     {
         List<SecondarySsrc> secondarySsrcs = new ArrayList<>();
-        for (SourceGroupPacketExtension sourceGroup : sourceGroups)
+        if (sourceGroups != null)
```

Comment:
instead of nesting an `if` here, can we do an early return in the case where it's null?

Reply:
also i think it may be better to fix this higher up the chain. in `org.jitsi.videobridge.RtpChannel.setRtpEncodingParameters(RtpChannel.java:1919)` we check for BOTH being empty or null, and if so return.  i think we should also check that if either is null, just replace it with an empty list.  then all the code in the calls from there can assume these are non-null (and would be even better to add a `@NotNull` annotation saying as such

Reply:
Sounds good.  Updated.  I don't see anywhere else in this project using @NotNull (which I believe is not a standard JDK thing, only used by some IDE e.g. JetBrains), so I'll keep it as is.

Reply:
We started to use the jetbrains @NotNull annotation recently and it's only used in a couple of places so far.